storagemarket/impl/provider.go

```diff
@@ -139,9 +139,8 @@ func (p *Provider) HandleDealStream(s network.StorageDealStream) {
 	err := p.receiveDeal(s)
 	if err != nil {
 		log.Errorf("%+v", err)
-		s.Close()
-		return
 	}
+	s.Close()
 }
 
 func (p *Provider) receiveDeal(s network.StorageDealStream) error {
```

Comment:
I  know it was in use before these changes, but I don't think any real purpose is being served by the receiveDeal abstraction. It's short and used only in HandleDealStream.